esphome/components/nextion/nextion.cpp

```diff
@@ -1034,9 +1057,42 @@ void Nextion::add_no_result_to_queue_with_command_(const std::string &variable_n
 
   if (this->send_command_(command)) {
     this->add_no_result_to_queue_(variable_name);
+#ifdef USE_NEXTION_COMMAND_SPACING
+  } else {
+    // Command blocked by spacing, add to queue WITH the command for retry
+    this->add_no_result_to_queue_with_pending_command_(variable_name, command);
+#endif  // USE_NEXTION_COMMAND_SPACING
   }
 }
 
+#ifdef USE_NEXTION_COMMAND_SPACING
+void Nextion::add_no_result_to_queue_with_pending_command_(const std::string &variable_name,
+                                                           const std::string &command) {
+#ifdef USE_NEXTION_MAX_QUEUE_SIZE
+  if (this->max_queue_size_ > 0 && this->nextion_queue_.size() >= this->max_queue_size_) {
+    ESP_LOGW(TAG, "Queue full (%zu), drop: %s", this->nextion_queue_.size(), variable_name.c_str());
+    return;
+  }
+#endif
+
+  ExternalRAMAllocator<nextion::NextionQueue> allocator(ExternalRAMAllocator<nextion::NextionQueue>::ALLOW_FAILURE);
```

Comment:
`ExternalRAMAllocator` is an alias as we renamed as it is not only external. Also the default constructor is fine and `ALLOW_FAILURE` does nothing but is kept around to keep the API compatible
```suggestion
  RAMAllocator<nextion::NextionQueue> allocator;
```

Reply:
I'm fine about changing this here, but this allocator is not part of this PR. And we do have the ExternalRAMAllocator in other parts of this component (which I'm happy to update - will put another PR for those).
Just a question here... Will the RAMAllocator also try PSRAM first?

Reply:
Ok, already found the answer t8 my questions on #7478

Reply:
> I'm fine about changing this here, but this allocator is not part of this PR

I see you have fixed it now, but these are new lines you added in this PR. 
I fixed the other ones in #9141